### ISSUE TITLE
Filter and sorting by video encoding 

### ISSUE THREAD

**Issue by toteto** [TheBeastLT/torrentio-scraper#95]
Considering all modern streaming devices support x265 HEVC encoding and the benefits it in offeres, offten times my preference for is to choose it ocer other encodings (like the x254).

Would be very usefull if there was option to filter by the video encoding (ex: I want only x265 and A1 encoded videos), and also include the encoding type in the "quality" qualifier (consider the 1080p x265 better quality then 10800 x264).

**Comment by TheBeastLT** [comment]
This is a very niche use case, which would complicate the addon configuration even more (most people using this addon don't even know what encoding is). I don't think that I'll be adding this feature, but I'll leave this open if there's more people wanting this.

**Comment by toteto** [comment]
You are right, it is indeed bit of a niche and advanced feature. But having such feature improves the streaming experience. The users are presented same high quality stream in almost half the size. 

The benefits are multifold, faster start of the stream, faster scrolling, lower network load.

As for the users, there can be "advanced" dropdown section in the configuration thar by default will have all the codecs selected.

Will look into the existint code for quality filtering and will try to implement the same for the encoding. Are you open for me submitting PR?

**Comment by TheBeastLT** [comment]
@toteto it's not about complexity adding such capability, it's about the complexity and questions/misconfigurations from users after adding such capability. It's because of that I don't think this would be valuable atm, so the same applies to any PRs.

**Comment by tundrablasen** [comment]
I would like to see this feature but i also understand @TheBeastLT view on that.

question: are you also maintaining the „lite“ version of torrentio which is preconfigured?

if yes then this could become the version 70% of Users are using. And we could get a „pro“ version where we can configure much more. Like file size limit, filter sorting video encoding, filter sorting audio e coding etc.

**Comment by Santhanam1** [comment]
> This is a very niche use case, which would complicate the addon configuration even more (most people using this addon don't even know what encoding is). I don't think that I'll be adding this feature, but I'll leave this open if there's more people wanting this.

If u think about it, This could be useful for all the non-RD user 
1080p (800MB) with 1000👤 and 
1080p (200MB) with 2👤 really hurts 

Yeah most people don't know about ENCODING But not all, atleast it improves 
2👤 -> 30👤(streamable)

And beleive me bro if anyone have to choose from 
1) 1080p (800MB) with 1000👤
2) 1080p (200MB) with 30👤

Most of people would choose (2)
Maybe We could reverse this.

**Comment by thebeast5618** [comment]
I also need this feature, though I agree it is niche. I see a decent amount of people are looking for this feature. I was even tinkering with the stremio-addon-sdk and found that you are absolutely right. 

"it's not about complexity adding such capability"

a dropdown "advanced video filtering" would really be a wonderful feature.

**Comment by dreamdenizen** [comment]
I don't think it's niche per se. There are a lot of devices that don't have AV1 Decode support, including all versions of the NVIDIA Shield. With more and more scene folks opting for AV1 now that more devices have dedicated encoders, I think it'll become a growing problem for a large swath of older devices out there. Anecdotaly, whenever I run into an AV1 file using the addon, Stremio crashes.

**Comment by rkr87** [comment]
I think this is something that should definitely be added, there are a bunch of devices that cannot decode H265.

Even something as simple as a free text comma-separated keyword exclusion list would be a drastic improvement.

**Comment by oxyisbad222** [comment]
it’s mostly an issue for Android box users and those who still are using old Fire devices. Sometimes those $20 nuggets can “technically” support 4K but not AV1 decode.
